src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -230,93 +190,138 @@ public function putAction($id, Request $request)
     {
-        return $this->get('sulu_core.doctrine_rest_helper');
+        $locale = $this->getRequestParameter($request, 'locale', true);
+        $data = [
+            'id' => $id,
+            'name' => (empty($request->get('name'))) ? null : $request->get('name'),
+            'key' => (empty($request->get('key'))) ? null : $request->get('key'),
+            'meta' => $request->get('meta'),
+            'parent' => $request->get('parent'),
+        ];
+        $category = $this->getManager()->save($data, $locale, $this->getUser()->getId(), $patch);
+
+        return $this->handleView($this->view($category, 200));
     }
 
     /**
-     * Handles the change of a category. Used in PUT and PATCH.
+     * Returns a category-list-representation for the category graph respective to the request.
+     * The category-list-representation contains only the root level of the category graph.
+     *
+     * If parentId is set, the root level of the sub-graph below the category with the given parentId is returned.
+     * If disableLimit is set, the list-representation ignores the limit of the request. This is used when querying
+     * for children of a category.
+     * If expandIds is set, the paths to the categories which are assigned to the ids are expanded. Categories are
+     * only expanded if they are descendants of the root level and the do not affect the limit or pagination.
      *
-     * @param $id
      * @param Request $request
+     * @param $locale
+     * @param null $parentId
+     * @param bool $disableLimit
+     * @param array $expandIds
      *
-     * @return \Symfony\Component\HttpFoundation\Response
+     * @return CategoryListRepresentation
      */
-    protected function saveEntity(Request $request, $id)
+    protected function getListRepresentation(Request $request, $locale, $parentId = null, $disableLimit = false, $expandIds = [])
     {
-        try {
-            $categoryManager = $this->get('sulu_category.category_manager');
-            $key = $request->get('key');
-            $locale = $this->getRequestParameter($request, 'locale', true);
-            $data = [
-                'id' => $id,
-                'key' => (empty($key)) ? null : $key,
-                'name' => $request->get('name'),
-                'meta' => $request->get('meta'),
-                'parent' => $request->get('parent'),
-                'locale' => $locale,
-            ];
-            $categoryEntity = $categoryManager->save($data, $this->getUser()->getId());
-            $categoryWrapper = $categoryManager->getApiObject($categoryEntity, $locale);
-
-            $view = $this->view($categoryWrapper, 200);
-        } catch (EntityNotFoundException $e) {
-            $view = $this->view($e->toArray(), 404);
-        } catch (KeyNotUniqueException $e) {
-            $view = $this->view($e->toArray(), 400);
+        $listBuilder = $this->initializeListBuilder($locale);
+
+        if (!$request->get('search')) {
```

Comment:
@danrot: usually we are just returning the root-level of the request-respective category tree, but it would be unintuitive to just return the results in the root level, when answering a search request.
i will add a comment here.